httpcore-nio/src/main/java/org/apache/http/impl/nio/codecs/IdentityDecoder.java

```diff
@@ -93,8 +93,9 @@ public long transfer(
 
         long bytesRead;
         if (this.buffer.hasData()) {
+            final int maxLen = this.buffer.length();
             dst.position(position);
-            bytesRead = this.buffer.read(dst);
+            bytesRead = this.buffer.read(dst, count < maxLen ? (int)count : maxLen);
```

Comment:
@suiryc I think simple long to int cast here may not be good enough. I suppose it should rather be `(int) Math.min(count, Integer.MAX_VALUE)`

Reply:
I thought that casting was enough because it is only done when `count < maxLen` where `maxLen` is already an `int` (and thus should make `count <= Integer.MAX_VALUE`).
In this case do you think it's better to make it more explicit as you suggest ? (and also in `LengthDelimitedDecoder`)

Reply:
@suiryc You are absolutely right. Disregard my comment.